pkg/api/types.go

```diff
@@ -421,6 +425,19 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
```

Comment:
That won't cross namespaces though, which means we probably either need this to be an object reference, or make this a configuration parameter to the kubelet instead (or take a DNS name)

----- Original Message -----
> > @@ -421,6 +425,19 @@ type NFSVolumeSource struct {
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> >  }
> >  
> > +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the
> > lifetime of a pod
> > +type GlusterfsVolumeSource struct {
> > +	// Required: EndpointsName is the endpoint name that details Glusterfs
> > topology
> > +	EndpointsName string `json:"endpoints"`
> 
> got it, thanks. would a special namespace for storage help?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6174/files#r27977053